### PR TITLE
Api error mgmt + enhanced provider

### PR DESCRIPTION
**Pull request by gzussa**
- Enhanced Client provider to easily support more clients in the future
- Improve Error management so that we can support distinct client errors in the future as well.

### CODE REVIEW

File: datadog/resource_datadog_dashboard_list.go [hunk 1]
```diff
@@ -49,102 +49,117 @@ func resourceDatadogDashboardList() *schema.Resource {
 }
 
 func resourceDatadogDashboardListCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	client := providerConf.CommunityClient
+
 	dashboardList, err := buildDatadogDashboardList(d)
 	if err != nil {
 		fmt.Printf("Error building the dashboard list %s", err.Error())
```
Comment (nmuesch) [inline datadog/resource_datadog_dashboard_list.go]: As discussed in person, we should return the error here.

File: datadog/resource_datadog_integration_pagerduty_service_object.go [hunk 2]
```diff
@@ -82,41 +84,46 @@ func resourceDatadogIntegrationPagerdutySORead(d *schema.ResourceData, meta inte
...
+	providerConf := meta.(*ProviderConfiguration)
+	client := providerConf.CommunityClient
 
 	_, err := client.GetIntegrationPDService(d.Id())
 	if err != nil {
 		if strings.Contains(err.Error(), "404 Not Found") {
 			return false, nil
 		}
-		return false, err
+		return false, translateClientError(err, "error getting pager duty integration service")
```
Comment (nmuesch) [inline datadog/resource_datadog_integration_pagerduty_service_object.go]: Nit, these should stay PagerDuty

Comment (nmuesch) [reply datadog/resource_datadog_integration_pagerduty_service_object.go]: Another nit, ,should we also say error checking exists to keep it consistent, like - https://github.com/terraform-providers/terraform-provider-datadog/pull/450/files#diff-f51916c63534715f41c1d444931ae717R370

File: datadog/resource_datadog_synthetics_test_.go [hunk 3]
```diff
@@ -203,7 +203,8 @@ func syntheticsTestOptions() *schema.Schema {
 }
 
 func resourceDatadogSyntheticsTestCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*datadog.Client)
+	providerConf := meta.(*ProviderConfiguration)
+	client := providerConf.CommunityClient
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_.go]: Nit: Should we print the same error messages in the create/update calls this resource makes?

File: datadog/resource_datadog_timeboard.go [hunk 4]
```diff
@@ -824,7 +824,9 @@ func resourceDatadogTimeboardCreate(d *schema.ResourceData, meta interface{}) er
 	if err != nil {
 		return fmt.Errorf("Failed to parse resource configuration: %s", err.Error())
 	}
-	timeboard, err = meta.(*datadog.Client).CreateDashboard(timeboard)
+	providerConf := meta.(*ProviderConfiguration)
+	client := providerConf.CommunityClient
+	timeboard, err = client.CreateDashboard(timeboard)
 	if err != nil {
 		return fmt.Errorf("Failed to create timeboard using Datadog API: %s", err.Error())
```
Comment (nmuesch) [inline datadog/resource_datadog_timeboard.go]: Should we use the new translateError method for these?

File: datadog/resource_datadog_synthetics_test_.go (hunk 3 above)
Comment (gzussa) [reply datadog/resource_datadog_synthetics_test_.go]: Good Catch!!

**Pull request merged**